### PR TITLE
added toast messages on facility and skills dialog

### LINKED ISSUE #7447: Missing Toast Messages for Success and Error in Linking/Unlinking Facility and Skill sections in users page
**Describe the bug**
Currently, when users attempt to link or unlink functionalities or skills, there is no visual confirmation of whether the action was successful or if an error occurred.

**To Reproduce**

1. Go to (https://care.ohc.network/users)
2. Click on Linked facility.
3. Perform actions like link, unlink facility.

**Expected behavior**

**Screenshots**

https://github.com/coronasafe/care_fe/assets/104725768/354eb591-e664-4a64-a109-609b7e3dba84

### PR DESCRIPTION
**Pull request by soumya-maheshwari**
## Proposed Changes

- Fixes #7447 
I've implemented toast messages for success and error events in the linking/unlinking facility and skill section in the users page

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## screenshots

https://github.com/coronasafe/care_fe/assets/104725768/1e6ba5ee-cfdd-4528-8d95-11e70a3181e2

### CODE REVIEW

File: src/Components/Users/ManageUsers.tsx [hunk 1]
```diff
@@ -650,11 +659,29 @@ function UserFacilities(props: { user: any }) {
         pathParams: { username },
       });
       if (res && res.status === 204) user.home_facility_object = null;
+      if (res?.status !== 204) {
+        Notification.Error({
```
Comment (rithviknishad) [inline src/Components/Users/ManageUsers.tsx]:
```suggestion
      if (!res?.ok) {
        user.home_facility_object = null;
        Notification.Error({
```

File: src/Components/Users/ManageUsers.tsx [hunk 2]
```diff
@@ -639,6 +639,15 @@ function UserFacilities(props: { user: any }) {
       body: { home_facility: facility.id.toString() },
     });
     if (res && res.status === 200) user.home_facility_object = facility;
+    if (res?.status !== 200) {
+      Notification.Error({
```
Comment (rithviknishad) [inline src/Components/Users/ManageUsers.tsx]:
```suggestion
    if (!res?.ok) {
      user.home_facility_object = facility;
      Notification.Error({
```

File: src/Components/Users/ManageUsers.tsx [hunk 3]
```diff
@@ -650,11 +659,29 @@ function UserFacilities(props: { user: any }) {
...
+          msg: "Home Facility cleared successfully",
+        });
+      }
     } else {
-      await request(routes.deleteUserFacility, {
+      const { res } = await request(routes.deleteUserFacility, {
         pathParams: { username },
         body: { facility: unlinkFacilityData?.facility?.id?.toString() },
       });
+      if (res?.status !== 204) {
```
Comment (rithviknishad) [inline src/Components/Users/ManageUsers.tsx]:
```suggestion
      if (!res?.ok) {
```

File: src/Components/Users/ManageUsers.tsx [hunk 4]
```diff
@@ -667,10 +694,17 @@ function UserFacilities(props: { user: any }) {
       pathParams: { username },
       body: { facility: facility.id.toString() },
     });
+
+    console.log(res);
+
```
Comment (rithviknishad) [inline src/Components/Users/ManageUsers.tsx]:
```suggestion
```

File: src/Components/Users/ManageUsers.tsx [hunk 5]
```diff
@@ -667,10 +694,17 @@ function UserFacilities(props: { user: any }) {
...
     if (res?.status !== 201) {
```
Comment (rithviknishad) [inline src/Components/Users/ManageUsers.tsx]:
```suggestion
    if (!res?.ok) {
```

File: src/Components/Users/SkillsSlideOver.tsx [hunk 6]
```diff
@@ -45,6 +45,9 @@ export default ({ show, setShow, username }: IProps) => {
         pathParams: { username },
         body: { skill: skill.id },
       });
+
+      console.log(res);
+
```
Comment (rithviknishad) [inline src/Components/Users/SkillsSlideOver.tsx]:
```suggestion
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Users/ManageUsers.tsx (hunk 2 above)
Comment (soumya-maheshwari) [reply src/Components/Users/ManageUsers.tsx]: Hi @rithviknishad,

I just wanted to confirm the proposed changes. It seems like user.home_facility_object = facility; should be moved to the else statement. If this adjustment aligns with our intentions, I can proceed with pushing the code.

Let me know if there are any concerns.

image

Comment (rithviknishad) [reply src/Components/Users/ManageUsers.tsx]: oh yes, you are right. my bad

Comment (soumya-maheshwari) [reply src/Components/Users/ManageUsers.tsx]: > oh yes, you are right. my bad

I've made some updates to the pull request:
Please review the changes and let me know if there are any further adjustments needed. Looking forward to your feedback!

**Review by rithviknishad — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**